chains/evm/contracts/pools/BurnMintFastTransferTokenPool.sol

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.10;
+
+import {IAny2EVMMessageReceiver} from "../interfaces/IAny2EVMMessageReceiver.sol";
+import {IFastTransferPool} from "../interfaces/IFastTransferPool.sol";
+import {IRMN} from "../interfaces/IRMN.sol";
+
+import {ITypeAndVersion} from "@chainlink/contracts/src/v0.8/shared/interfaces/ITypeAndVersion.sol";
+import {IBurnMintERC20} from "@chainlink/contracts/src/v0.8/shared/token/ERC20/IBurnMintERC20.sol";
+import {IERC165} from
+  "@chainlink/contracts/src/v0.8/vendor/openzeppelin-solidity/v4.8.3/contracts/utils/introspection/IERC165.sol";
+
+import {BurnMintTokenPoolAbstract} from "./BurnMintTokenPoolAbstract.sol";
+import {FastTransferTokenPoolAbstract} from "./FastTransferTokenPoolAbstract.sol";
+import {TokenPool} from "./TokenPool.sol";
+
+// OpenZeppelin imports
+import {IERC20} from
+  "@chainlink/contracts/src/v0.8/vendor/openzeppelin-solidity/v4.8.3/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from
+  "@chainlink/contracts/src/v0.8/vendor/openzeppelin-solidity/v4.8.3/contracts/token/ERC20/utils/SafeERC20.sol";
+
+contract BurnMintFastTransferTokenPool is ITypeAndVersion, BurnMintTokenPoolAbstract, FastTransferTokenPoolAbstract {
+  using SafeERC20 for IERC20;
+
+  string public constant override typeAndVersion = "BurnMintFastTransferTokenPool 1.6.1";
+
+  constructor(
+    IBurnMintERC20 token,
+    uint8 localTokenDecimals,
+    address[] memory allowlist,
+    address rmnProxy,
+    address router
+  )
+    // FastTransferTokenPoolAbstract.LaneConfigArgs[] memory laneConfigArgs
+    FastTransferTokenPoolAbstract(token, localTokenDecimals, allowlist, rmnProxy, router)
+  {}
+
+  function _handleTokenToTransfer(uint64 destinationChainSelector, address sender, uint256 amount) internal override {
+    if (IRMN(i_rmnProxy).isCursed(bytes16(uint128(destinationChainSelector)))) revert CursedByRMN();
```

Comment:
since these checks and rate limit consumption will most likely occur in all fast transfer token pools I would prefer this be moved into the abstract parent.